src/erlcloud_config.erl

```diff
@@ -0,0 +1,632 @@
+    end.
+
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% GetResourceConfigHistory
+%%
+%% https://docs.aws.amazon.com/config/latest/APIReference/API_GetResourceConfigHistory.html
+%%------------------------------------------------------------------------------
+-spec get_resource_config_history() ->
+    result_paged(map()).
+get_resource_config_history() ->
+    get_resource_config_history(#{}).
+
+-spec get_resource_config_history(Opts :: map()) ->
+    result_paged(map()).
+get_resource_config_history(Opts) ->
+    get_resource_config_history(Opts, default_config()).
+
+-spec get_resource_config_history(Opts :: map(),
+    Config :: aws_config()) ->
+    result_paged(map()).
+get_resource_config_history(Opts, Config) when is_map(Opts) orelse Opts =:= #{} ->
+    Request = #{?CHRONOLOGICAL_ORDER => maps:get(?CHRONOLOGICAL_ORDER, Opts, null),
+        ?EARLIER_TIME                => maps:get(?EARLIER_TIME, Opts, null),
+        ?LATER_TIME                  => maps:get(?LATER_TIME, Opts, null),
+        ?LIMIT                       => maps:get(?LIMIT, Opts, ?LIMIT_MAX),
+        ?NEXT_TOKEN_LABEL            => maps:get(?NEXT_TOKEN_LABEL, Opts, <<"">>),
+        ?RESOURCE_ID                 => maps:get(?RESOURCE_ID, Opts, null),
+        ?RESOURCE_TYPE               => maps:get(?RESOURCE_TYPE, Opts, null)},
+    case request(Config, "GetResourceConfigHistory", Request) of
+        {ok, Res} ->
+            {ok, Res, get_next_token(Res)};
+        {error,  _} = E -> E
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% ListDiscoveredResources
+%%
+%% https://docs.aws.amazon.com/config/latest/APIReference/API_ListDiscoveredResources.html
+%%------------------------------------------------------------------------------
+-spec list_discovered_resources() ->
+    result_paged(map()).
+list_discovered_resources() ->
+    list_discovered_resources(#{}).
+
+-spec list_discovered_resources(Opts :: map()) ->
+    result_paged(map()).
+list_discovered_resources(Opts) ->
+    list_discovered_resources(Opts, default_config()).
+
+-spec list_discovered_resources(Opts :: map(),
+    Config :: aws_config()) ->
+    result_paged(map()).
+list_discovered_resources(Opts, Config) when is_map(Opts) orelse Opts =:= #{} ->
+    Request = #{?INCLUDE_DELETED_RESOURCES => maps:get(?INCLUDE_DELETED_RESOURCES, Opts, true),
+        ?LIMIT                             => maps:get(?LIMIT, Opts, ?LIMIT_MAX),
+        ?NEXT_TOKEN_LABEL                  => maps:get(?NEXT_TOKEN_LABEL, Opts, <<"">>),
+        ?RESOURCE_IDS                      => maps:get(?RESOURCE_IDS, Opts, []),
+        ?RESOURCE_NAME                     => maps:get(?RESOURCE_NAME, Opts, null),
+        ?RESOURCE_TYPE                     => maps:get(?RESOURCE_TYPE, Opts, null)},
+    case request(Config, "ListDiscoveredResources", Request) of
+        {ok, Res} ->
+            {ok, Res, get_next_token(Res)};
```

Comment:
I just followed the README. I have no preference either way.

Reply:
I think README has  a bit of ambiguity on that, as it's just says about preferring one way to another, and the third way for records...

I'm up for `{ok, Res}` approach rather then `{ok, Results, Marker}` because it will be up to customer whether he wants to actually retrieve the token or not, so we don't do excessive action.